newrelic/resource_newrelic_workload.go

```diff
@@ -0,0 +1,327 @@
+		Computed: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"email": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"gravatar": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+				"id": {
+					Type:     schema.TypeInt,
+					Computed: true,
+				},
+				"name": {
+					Type:     schema.TypeString,
+					Computed: true,
+				},
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceNewRelicWorkloadCreate,
+		Read:   resourceNewRelicWorkloadRead,
+		Update: resourceNewRelicWorkloadUpdate,
+		Delete: resourceNewRelicWorkloadDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"account": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: "The account the workload belongs to.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"created_at": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The moment when the object was created, represented in milliseconds since the Unix epoch.",
+			},
+			"created_by": userReference,
+			"entity": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Computed:    true,
+				Description: "A list of entities manually assigned to this workload.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"guid": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "The unique entity identifier in New Relic.",
+						},
+					},
+				},
+			},
+			"entity_search_query": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "A list of search queries that define a dynamic workload.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of the query.",
+						},
+						"query": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The query.",
+						},
+					},
+				},
+			},
+			"guid": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique entity identifier of the workload in New Relic.",
+			},
+			"workload_id": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The unique entity identifier of the workload.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The workload's name.",
+			},
+			"permalink": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL of the Workload.",
+			},
+			"scope_accounts": {
```

Comment:
Similar to the above, we should be able to simplify this to look like 
```
"storage_account_ids": {
  Type:     schema.TypeSet,
  Optional: true,
  Elem: &schema.Schema{
  Type: schema.TypeString,
},
```

Reply:
👍